ims/einstein/iscsi/tgt.py

```diff
@@ -101,57 +111,102 @@ def add_target(self, target_name):
         :param target_name: Name of target to be added
         :return: None
         """
-        try:
-            targets = self.list_targets()
-            if target_name not in targets:
-                self.__generate_config_file(target_name)
-                command = "tgt-admin --execute"
-                shell.call(command, sudo=True)
-            else:
-                raise iscsi_exceptions.TargetExistsException()
-        except (IOError, OSError) as e:
-            raise iscsi_exceptions.TargetCreationFailed(str(e))
-        except shell_exceptions.CommandFailedException as e:
-            raise iscsi_exceptions.TargetCreationFailed(str(e))
+        targets = self.list_targets()
+        if target_name in targets:
+            raise iscsi_exceptions.TargetExistsException()
+
+        self.__generate_config_file(target_name)
+
+        command = "sudo tgt-admin --execute"
```

Comment:
Why do we need to `sudo` now when we didn't before?

Reply:
because earlier the code had to be run as root...by specifying sudo I think we can make an entry in sudoers file for the bmi-user to only execute tgt-admin commands. I am open to other ideas if there's a better way